enhancements/installer/image-based-installer.md

```diff
@@ -0,0 +1,255 @@
+during the generation of the ISO image.
+
+The image-based installer offers key advantages, where fast and reliable deployment at the edge is crucial.
+By generating an ISO images containing all necessary components, the image-based installer significantly accelerates
+deployment times. Moreover, unlike the agent-based installer, the image-based approach allows for cluster configuration
+to be supplied upon deployment at the edge, rather than during the initial ISO generation process. 
+This flexibility enables operators to use a single generic image for installing multiple clusters, streamlining the
+deployment process and reducing the need for multiple customized ISO images.
+
+The command-line tool will also support generating a configuration ISO with all the site specific configuration data for
+the cluster to be installed provided as input.
+The Config Image Contents:
+* ClusterInfo (cluster name, base domain, hostname, nodeIP)
+* SSH authorized_keys
+* Pull Secret
+* Extra Manifests
+* Generated keys and certs (compatible the generated admin kubeconfig)
+* Static networking config
+
+The site specific configuration data will be generated according to information provided
+in the install-config.yaml and the manifests provided in the installation directory as input.
+To complete the installation at the edge site:
+The cluster configuration for the edge location can be delivered by copying the config iso content onto the node
+and placing it under /opt/openshift/cluster-configuration/
+The cluster configuration can also b delivered using an attached ISO, a systemd service running on the host
+pre-installed Image-based Installer will mount that ISO (identified by known label) and copy the cluster configuration
+to /opt/openshift/cluster-configuration/
+The cluster configuration data on the disk will be used to configure the cluster and allow OCP to start successfully.
+
+### Workflow Description
+
+TBD
+
+### API Extensions
+
+N/A
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+The image-based installer targets single-node OpenShift deployments.
+
+### Implementation Details/Notes/Constraints
+
+Since we must allow users to provision hosts themselves, either manually or
+using automated tooling of their choice, the ISO format offers the widest range
+of compatibility. Building a single ISO to boot multiple hosts makes it
+considerably easier for the user to manage. The additional site configuration
+ISO is necessary for configuring each cluster securely and independently.
+
+The user, before running the image-based installer, must generate a [seed image](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/seed-image-generation.md)
+via the [Lifecycle Agent SeedGenerator Custom Resouce (CR)](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/seed-image-generation.md).
```

Comment:
Is pushing the seed image to a container registry an integral part of the build process, or just a way to share it with the process building the ISO?

It seems to me that you could cut out the middleman and have this operator generate an ISO directly. (It could serve it over HTTP and put the URL in the Status.)

The base ISO is already part of the release payload of the running cluster, and the user's pull secret is present in the cluster. You can reduce the amount of configuration required, eliminate a step, and rule out any possibility of version skew.

Reply:
The Lifecycle Agent SeedGenerator is generating a seed OCI image to be used for image-based upgrade.
We can consider another CR with all the fields required for generating the installation ISO and get seed image or reference SeedGenerator and generate an installation ISO + expose the download URL.
I really like the idea of letting the controller do it instead of asking to user to obtain and execute the CLI, however this will mean that a user will need to have a running openshift with LCA to get the IBI installation ISO and that  the user will need to move around the ISO file. while if we expose the ISO generation functionality in the openshift-installer the user that needs the ISO just need the installer binary and a config file

Reply:
@zaneb I am not fully sure i understand comment regarding installation disk. 
It will be used to write ostree into it and should not be dependent on seed.

Reply:
1. Pull secret is removed from seed image and there is not requirement that seed image was created by the same user as target. The idea is that they will be totally different users.
2. Same as 1

Reply:
But in short IBI has 3 stages that can be run by different personas and thats why we need to provide creds each time.
1. Create seed, it will be ran somewhere, we are clearing everything that is relevant for user configuration like pull secret and some other things
2. Installation iso: this is very similar to infraenv stage in assisted, user provides configuration in order to write seed image to disk and to precache images. After that machine can be sent to vendor client for example.
3. Configuration iso : those will be params that we will configure for the edge device owner when he boot the device, those params will be configured on the cluster (without need reboot it)

Reply:
OK, let's make sure this is documented.